src/components/ParserOpenRPC/RequestBox/index.tsx

```diff
@@ -48,21 +65,20 @@ export default function RequestBox({
           {params.length > 0 && (
             <button
               className={clsx(global.linkBtn, "margin-right--md")}
-              disabled={!isMetamaskInstalled}
+              disabled={isRunAndCustomizeRequestDisabled}
               onClick={openModal}
               data-test-id="customize-request"
             >
               Customize request
             </button>
           )}
-          <button
-            className={global.primaryBtn}
-            disabled={!isMetamaskInstalled}
-            onClick={submitRequest}
-            data-test-id="run-request"
-          >
-            Run request
-          </button>
+          {
+            isRunAndCustomizeRequestDisabled ?
+              (<Tooltip message="Before you can run or customize requests, please connect your MetaMask wallet first.">
```

Comment:
```suggestion
              (<Tooltip message="Before you can run or customize this request, please connect your MetaMask wallet first.">
```

Suggestion to clarify that this blocker is only for the specific method(s).